packages/xo-web/src/xo-app/vm/tab-advanced.js

```diff
@@ -288,6 +318,132 @@ class CoresPerSocket extends Component {
   }
 }
 
+class AddAclsModal extends Component {
+  get value() {
+    return this.state
+  }
+
+  _getPredicate = createSelector(
+    () => this.props.acls,
+    () => this.props.vm,
+    (acls, { id: object }) => ({ id: subject }) =>
+      !some(acls, { object, subject })
+  )
+
+  render() {
+    const { action, subjects } = this.state
+    return (
+      <form>
+        <div className='form-group'>
+          <SelectSubject
+            multi
+            onChange={this.linkState('subjects')}
+            predicate={this._getPredicate()}
+            value={subjects}
+          />
+        </div>
+        <div className='form-group'>
+          <SelectRole onChange={this.linkState('action')} value={action} />
+        </div>
+      </form>
+    )
+  }
+}
+
+const Acls = decorate([
+  addSubscriptions({
+    acls: subscribeAcls,
+    groups: cb => subscribeGroups(groups => cb(keyBy(groups, 'id'))),
+    users: cb => subscribeUsers(users => cb(keyBy(users, 'id'))),
+  }),
+  provideState({
+    effects: {
+      addAcls: () => (state, { acls, vm }) =>
+        confirm({
+          title: _('vmAddAcls'),
+          body: <AddAclsModal acls={acls} vm={vm} />,
+        })
+          .then(({ action, subjects }) => {
+            if (action == null || isEmpty(subjects)) {
+              throw new Error('User(s)/group(s) and role are required.')
+            }
+            return (
+              Promise.all(
+                map(subjects, subject =>
+                  addAcl({ subject, object: vm, action })
+                )
+              ),
+              noop
+            )
+          })
+          .catch(err => err && error('Add ACL(s)', err.message || String(err))),
+    },
+    computed: {
+      vmAcls: (_, { acls, groups, users, vm }) => {
+        if (isEmpty(acls)) {
+          return
+        }
+        const vmAcls = []
+        acls.forEach(({ object, subject, action }) => {
+          if (object === vm.id) {
+            vmAcls.push({
+              action,
+              subject: users[subject] || groups[subject],
+            })
+          }
+        })
+        return vmAcls
+      },
+    },
+  }),
+  injectState,
+  ({ state: { vmAcls, vmId }, effects, vm }) => {
+    return (
+      <Container>
+        <Row>
+          {vmAcls !== undefined &&
+            vmAcls.slice(0, 5).map(({ subject, action }) => (
+              <Col
+                key={Math.random()
```

Comment:
By doing this, you're basically telling React that on each render, the component is a different one. So React will create a new one (new mount, new lifecycle, etc.).
What you can do though is ``key={`${subject.email}.${action}`}``.